src/rules/spaceInParensRule.ts

```diff
@@ -0,0 +1,231 @@
+        `,
+      Lint.Utils.dedent`
+        "${RULE_NAME}": [true, "always", { "exceptions": [ "{}", "[]", "()", "empty" ] }]
+        `
+    ],
+    typescriptOnly: false,
+    type: 'style'
+  };
+
+  public static MISSING_SPACE_MESSAGE = 'there must be a space inside this paren.';
+  public static REJECTED_SPACE_MESSAGE = 'there should be no spaces inside this paren.';
+
+  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+    const walker = new SpaceInParensWalker(sourceFile, this.getOptions());
+    return this.applyWithWalker(walker);
+  }
+}
+
+class SpaceInParensWalker extends Lint.RuleWalker {
+  // ** RULE IMPLEMENTATION HERE **
+  private spaced: boolean;
+  private exceptionsArrayOptions = [];
+  private braceException: boolean;
+  private bracketException: boolean;
+  private parenException: boolean;
+  private empty: boolean;
+
+  constructor(sourceFile: ts.SourceFile, options: Lint.IOptions) {
+    super(sourceFile, options);
+    const ruleOptions = this.getOptions();
+    this.spaced = this.hasOption(ALWAYS) || (ruleOptions && ruleOptions.length === 0);
+
+    if (ruleOptions[1]) {
+      this.exceptionsArrayOptions = (ruleOptions.length === 2) ? ruleOptions[1].exceptions : [] ;
+      if (this.exceptionsArrayOptions.length) {
+        this.braceException = this.exceptionsArrayOptions.indexOf('{}') !== -1;
+        this.bracketException = this.exceptionsArrayOptions.indexOf('[]') !== -1;
+        this.parenException = this.exceptionsArrayOptions.indexOf('()') !== -1;
+        this.empty = this.exceptionsArrayOptions.indexOf('empty') !== -1;
+      }
+    }
+  }
+
+  private getExceptions() {
+    const openers = [];
+    const closers = [];
+
+    if (this.braceException) {
+      openers.push(ts.SyntaxKind.OpenBraceToken);
+      closers.push(ts.SyntaxKind.CloseBraceToken);
+    }
+
+    if (this.bracketException) {
+      openers.push(ts.SyntaxKind.OpenBracketToken);
+      closers.push(ts.SyntaxKind.CloseBracketToken);
+    }
+
+    if (this.parenException) {
+      openers.push(ts.SyntaxKind.OpenParenToken);
+      closers.push(ts.SyntaxKind.CloseParenToken);
+    }
+
+    if (this.empty) {
+      openers.push(ts.SyntaxKind.CloseParenToken);
+      closers.push(ts.SyntaxKind.OpenParenToken);
+    }
+
+    return {
+      openers,
+      closers
+    };
+  }
+
+  protected findParenNodes(node: ts.Node): ts.Node[] {
+    const children = node.getChildren();
+    let openParneNode: ts.Node;
+    let secondOne: ts.Node;
+    let oneBeforLast: ts.Node;
+    let closeParanNode: ts.Node;
+    for (let i = 0; i < children.length; i++) {
+      if (children[i].kind === ts.SyntaxKind.OpenParenToken) {
+        openParneNode = children[i];
+        secondOne = children[i + 1];
+      }
+      if (children[i].kind === ts.SyntaxKind.CloseParenToken) {
+        oneBeforLast = children[i - 1];
+        closeParanNode = children[i];
+      }
+    }
+    return [ openParneNode, secondOne, oneBeforLast, closeParanNode ];
```

Comment:
Try to keep variable names consistent, use `paren` for paren related variables. Also try names like `firstChild, lastChild` instead of `secondOne, oneBeforeLast`.